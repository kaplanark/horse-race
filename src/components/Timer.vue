<script setup>
import { computed } from 'vue';
const props = defineProps({
	time: {
		type: Number,
		default: 0
	}
});

const timer = computed(() => {
	let minutes = Math.floor(props.time / 60000); // calculating the minutes
	let seconds = ((props.time % 60000) / 1000).toFixed(0); // calculating the seconds
	return minutes + ":" + (seconds < 10 ? '0' : '') + seconds; // return the time in the format of minutes:seconds
});
</script>

<template>
	<div class="timer">{{ timer }}</div>
</template>

<style lang="scss">
.timer {
	font-size: 24px;
	font-family: var(--font-secondary);
	color: var(--color-primary);
}
</style>

<style lang="scss">
.timer {
	font-size: 24px;
	font-family: var(--font-secondary);
	display: flex;
	justify-content: center;
	align-items: center;
}
</style>