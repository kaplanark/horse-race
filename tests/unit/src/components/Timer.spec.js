import { describe, expect, test } from 'vitest'
import { mount } from '@vue/test-utils'
import Timer from '@components/Timer.vue';

describe('Timer.vue', () => {
	let wrapper;
	beforeEach(() => {
		wrapper = mount(Timer, {
			props: {
				time: 1000,
			}
		});
	});

	afterEach(() => {
		wrapper.unmount();
	});

	test('is correctly rendered', () => {
		expect(wrapper.html()).toMatchSnapshot();
	});

	test('should renders is page content is correct', () => {
		expect(wrapper.vm).toBeTruthy();
	});

	test('renders props.time when passed', () => {
		expect(wrapper.text()).toBe('0:01');
	});
});