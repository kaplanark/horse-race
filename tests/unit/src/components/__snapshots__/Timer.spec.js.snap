// Vitest Snapshot v1

exports[`Timer.vue > is correctly rendered 1`] = `"<div class=\\"timer\\">0:01</div>"`;
